### DISCUSSION TITLE
Missing Device: Aqara knob H1 (with Neutral) ZNXNKG01LM

### DISCUSSION THREAD

**Discussion by Piksanhome** [Koenkk/zigbee2mqtt#17653] [Devices]
Hi,

I have a new device, Aqara knob H1 (with Neutral). The zigbee ID is showing: **lumi.switch.rkna01** and model number is: **ZNXNKG01LM** I find that z2m is just supporting a similar product with same outlook and function, but that one is in wireless. 
https://www.zigbee2mqtt.io/devices/ZNXNKG02LM.html

Is there any way to copy the **ZNXNKG02LM** config and apply to the missing one?

**Comment by mageali** [comment]
FIRSTLY, I HAVE NO IDEA WHAT I'M DOING, I JUST COPY-PASTED SOME CODE (ZNXNKG02LM) AND MADE MINOR MODIFICATIONS. HAVE SOMEONE GO OVER IT AND HOPEFULLY IMPROVE IT OR USE IT AT YOUR OWN RISK.

That said, I made it work as far as I can tell, but with some limitations, the power (wattage) is not being reported correctly, and I couldn't get the mode to be coupled/decoupled or the mode to be event/command. You can control the 3 relays independently and it will read the knob rotation angle, speed, percentage...etc

Any idea how I can get this improved/fixed/revised and maybe eventually published. My first time on Github, so this whole thing is very new to me?

    {
        zigbeeModel: ['lumi.switch.rkna01'],
        model: 'ZNXNKG01LM',
        vendor: 'Xiaomi',
        description: 'Aqara knob H1 (with Neutral)',
        meta: {multiEndpoint: true},
        endpoint: (device) => {
            return {'left': 1, 'center': 2, 'right': 3};
        },
        exposes: [e.switch().withEndpoint('left'), e.switch().withEndpoint('center'), e.switch().withEndpoint('right'), e.device_temperature(),
        e.action(['single_left', 'double_left', 'single_center', 'double_center', 'single_right', 'double_right', 'single_both', 'double_both']),
        e.power_outage_count(), e.energy(), e.voltage(), e.power(),
        exposes.numeric('action_rotation_angle', ea.STATE).withUnit('*').withDescription('Rotation angle'),
        exposes.numeric('action_rotation_angle_speed', ea.STATE).withUnit('*').withDescription('Rotation angle speed'),
        exposes.numeric('action_rotation_percent', ea.STATE).withUnit('%').withDescription('Rotation percent'),
        exposes.numeric('action_rotation_percent_speed', ea.STATE).withUnit('%').withDescription('Rotation percent speed'),
        exposes.numeric('action_rotation_time', ea.STATE).withUnit('ms').withDescription('Rotation time'),
        ],
        fromZigbee: [fz.aqara_opple, fz.aqara_knob_rotation, fz.xiaomi_power, fz.on_off],
        toZigbee: [tz.aqara_opple_operation_mode, tz.on_off, tz.xiaomi_switch_operation_mode_opple],
        onEvent: preventReset,
    },

**Reply by sozoalvin** [reply]
@mageali 
dear mageali, can i check with you; how do you get all 3 gangs to be controlled independently using your config? 

for e.g. if you can click the wheel button once to have L1 turned on, what are the things you need to "press/do" on the switch in order to have L1, L2, L3 be independently controlled? 

thank you in advance for your sharing!

**Reply by mariodivece** [reply]
I just bought one of these and I can't figure out how to control the 3 relays independently. I am working on a converter. Will share some findings with you briefly. I have it "mostly" working but still figuring out additional stuff.

**Reply by mageali** [reply]
@sozoalvin single press automatically toggles the 3 relays, and I couldn't get it to run decoupled to override that. l1,l2,l3 can be toggled independently through the interface (aqara home app or z2m in this case) you also have double click an rotation, but as Mario mentioned below, this is not a TRIAC dimmer, its just a nice looking knob, you would write your code to dim a smart bulb based on the rotation angle (or smart dimmer I guess would work also)

**Comment by mariodivece** [comment]
Here's some progress... still a lot to figure out though. @mageali this is my first time trying to create a converter and I am learning a lot from your code. Thanks for sharing!

```js
const { } = require('zigbee-herdsman-converters/lib/modernExtend');

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const xiaomi = require('zigbee-herdsman-converters/lib/xiaomi');
const e = exposes.presets;
const ea = exposes.access;

const manufacturerCode = xiaomi.manufacturerCode;

const preventReset = async (type, data, device) => {
    if (
        // options.allow_reset ||
        type !== 'message' ||
        data.type !== 'attributeReport' ||
        data.cluster !== 'genBasic' ||
        !data.data[0xfff0] ||
        // eg: [0xaa, 0x10, 0x05, 0x41, 0x87, 0x01, 0x01, 0x10, 0x00]
        !data.data[0xFFF0].slice(0, 5).equals(Buffer.from([0xaa, 0x10, 0x05, 0x41, 0x87]))
    ) {
        return;
    }
    const payload = {
        [0xfff0]: {
            value: [0xaa, 0x10, 0x05, 0x41, 0x47, 0x01, 0x01, 0x10, 0x01],
            type: 0x41,
        }
    };
    await device.getEndpoint(1).write('genBasic', payload, { manufacturerCode });
};

/*
Zigbee2MQTT:debug 2024-01-24 18:06:24: Received Zigbee message from '0x54ef4410003e6e63', type 'attributeReport', cluster 'aqaraOpple', data '{"247":{"data":[160,32,3,100,16,1,101,16,1,102,16,1,3,40,33,152,57,0,0,0,0,149,57,0,0,0,0,150,57,179,246,154,68,151,57,0,0,0,0,5,33,1,0,154,32,0,9,33,0,3,11,32,0,13,35,23,15,0,0],"type":"Buffer"}}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2024-01-24 18:06:24: ZNXNKG01LM: Processed buffer into data {
    "3":33, // device_temperature
    "5":1, // power_outage_count
    "9":768, // ??
    "11":0, // illuminance ?? 
    "13":3863, // ??
    "100":1, // state_left
    "101":1, // state_center
    "102":1, // state_right
    "149":0, // energy(opts)/consumption
    "150":1239.7093505859375, // voltage * 100
    "151":0, // current (Amps?)
    "152":0, // power (W?)
    "154":0, // ?
    "160":3} // gas/smoke? (unlikely)
*/

const definition = {
    zigbeeModel: ['lumi.switch.rkna01'], // as opposed to wireless 'lumi.remote.rkba01'
    model: 'ZNXNKG01LM', // as opposed to wireless ZNXNKG02LM
    vendor: 'Xiaomi',
    description: 'Aqara Smart Knob H1 (With Neutral)',
    meta: { multiEndpoint: true }, // flag the device needs distinct cluster endpoints
    endpoint: (device) => {
        // name the cluster endpoints (genOnOff for endpoint 1,2,3)
        return {
            'left': 1,
            'center': 2,
            'right': 3
        };
    },
    exposes: [

        e.device_temperature(),
        e.power_outage_count(),
        e.energy(),
        e.voltage(),
        e.power(),

        // fz.on_off and tz.on_off
        e.switch().withEndpoint('left').withDescription('Relay 1'),
        e.switch().withEndpoint('center').withDescription('Relay 2'),
        e.switch().withEndpoint('right').withDescription('Relay 3'),

        e.action([
            // coming from fz aqara_knob_rotation
            'start_rotating', 'rotation', 'stop_rotating',

            // unfortunately, if we want individual relay control, we need to suffix the actions with the name of the endpoint
            // we previously defined :(
            'single', 'double', 'hold', 'release',
            'single_left', 'double_left',
            'single_center', 'double_center',
            'single_right', 'double_right',
            'single_both', 'double_both']),

        // state coming from fz.aqara_knob_rotation
        e.numeric('action_rotation_angle', ea.STATE).withUnit('*').withDescription('Rotation angle'),
        e.numeric('action_rotation_angle_speed', ea.STATE).withUnit('*').withDescription('Rotation angle speed'),
        e.numeric('action_rotation_percent', ea.STATE).withUnit('%').withDescription('Rotation percent'),
        e.numeric('action_rotation_percent_speed', ea.STATE).withUnit('%').withDescription('Rotation percent speed'),
        e.numeric('action_rotation_time', ea.STATE).withUnit('ms').withDescription('Rotation time'),

        // try relay and decoupled? maybe unsupported?
        e.enum('operation_mode', ea.ALL, ['control_relay', 'decoupled']).withDescription('Decoupled mode'),
    ],
    fromZigbee: [
        fz.on_off, // will get basic on/off info from clusters named genOnOff (relays)
        fz.xiaomi_on_off_action, // single, double, (also has hold, release defined)
        fz.aqara_knob_rotation, // will map to aqaraOpple actions start_rotating, rotation, stop_rotating
        xiaomi.fromZigbee.aqara_opple, // will decode aqaraOpple data (including power)
        fz.ignore_metering, // rely on aqaraOpple instead of cluster haElectricalMeasurement
        fz.ignore_electrical_measurement, // rely on aqaraOpple instead of cluster haElectricalMeasurement
    ],
    toZigbee: [
        tz.on_off, // control individual relays
        tz.xiaomi_switch_operation_mode_opple // TODO
    ],
    onEvent: preventReset
};

module.exports = definition;
```

**Reply by mariodivece** [reply]
Here are the related issues I've found so far (for my own notes)
 - #13290 
 - #14888 
 - [Might find something useful here](https://github.com/niceboygithub/AqaraGateway/issues/14)

**Comment by mariodivece** [comment]
Okay, back from my travels. I spent the better part of this morning debugging this device. In short, and IMO, it's not a very good device.
It's a "dimmer" but it has no triac dimmer inside. It's just a knob with 3 internal relays which makes little to no sense. It's aesthetically pleasing but that's about where it ends. Does not seem to support decoupled mode either. Total disappointment. Here's the converter that seems to work for the most part:

```js
const { } = require('zigbee-herdsman-converters/lib/modernExtend');

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const xiaomi = require('zigbee-herdsman-converters/lib/xiaomi');

const e = exposes.presets;
const ea = exposes.access;

const manufacturerCode = xiaomi.manufacturerCode;

const preventReset = async (type, data, device) => {
    if (
        // options.allow_reset ||
        type !== 'message' ||
        data.type !== 'attributeReport' ||
        data.cluster !== 'genBasic' ||
        !data.data[0xfff0] ||
        // eg: [0xaa, 0x10, 0x05, 0x41, 0x87, 0x01, 0x01, 0x10, 0x00]
        !data.data[0xFFF0].slice(0, 5).equals(Buffer.from([0xaa, 0x10, 0x05, 0x41, 0x87]))
    ) {
        return;
    }
    const payload = {
        [0xfff0]: {
            value: [0xaa, 0x10, 0x05, 0x41, 0x47, 0x01, 0x01, 0x10, 0x01],
            type: 0x41,
        }
    };
    await device.getEndpoint(1).write('genBasic', payload, { manufacturerCode });
};

const definition = {
    zigbeeModel: ['lumi.switch.rkna01'], // as opposed to wireless 'lumi.remote.rkba01'
    model: 'ZNXNKG01LM', // as opposed to wireless ZNXNKG02LM
    vendor: 'Xiaomi',
    description: 'Aqara Smart Knob H1 (With Neutral)',
    endpoint: (device) => {
        return {'left': 1, 'center': 2, 'right': 3};
    },
    options: [exposes.options.state_action(false)],
    meta: {multiEndpoint: true},
    exposes: [
        // the 3 internal relays
        e.switch().withEndpoint('left'),
        e.switch().withEndpoint('center'),
        e.switch().withEndpoint('right'),

        // knob actions and states
        e.action([
            'single', 'double', 'hold', 'release',
            'start_rotating', 'rotation', 'stop_rotating'
        ]),
        e.numeric('action_rotation_angle', ea.STATE).withUnit('*').withDescription('Rotation angle'),
        e.numeric('action_rotation_angle_speed', ea.STATE).withUnit('*').withDescription('Rotation angle speed'),
        e.numeric('action_rotation_percent', ea.STATE).withUnit('%').withDescription('Rotation percent'),
        e.numeric('action_rotation_percent_speed', ea.STATE).withUnit('%').withDescription('Rotation percent speed'),
        e.numeric('action_rotation_time', ea.STATE).withUnit('ms').withDescription('Rotation time'),

        // power and temperature
        e.power(),
        e.current(),
        e.energy(),
        e.voltage(),
        e.device_temperature(),
        e.power_outage_count(),
    ],
    fromZigbee: [
        fz.on_off,
        fz.xiaomi_power,
        fz.xiaomi_multistate_action,
        fz.aqara_knob_rotation,
        xiaomi.fromZigbee.aqara_opple,
    ],
    toZigbee: [
        tz.on_off,
    ],
    onEvent: preventReset,
    configure: async (device, coordinatorEndpoint, logger) => {
        await device.getEndpoint(1).write('aqaraOpple', {'mode': 1}, {manufacturerCode: 0x115f, disableResponse: true});
    },
};

module.exports = definition;
```

**Reply by sozoalvin** [reply]
hey @mariodivece thanks for sharing the code - i do want to ask. at this moment using your converter; what works and what doesnt?

**Comment by mageali** [comment]
@mariodivece Great work, I've tidied up my version of the convertor since my first post, but more or less it does what your convertor does here

@Koenkk I've never done a pull request, any guidance how we can get this convertor added to Z2M Convertors?

**Reply by Koenkk** [reply]
Press the edit button [here](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/lumi.ts) (pencil)

**Reply by mageali** [reply]
@Koenkk Thanks for the help so far. In the previous I had the convertor partially working, but still had a couple of bugs to fix, after the latest update it's not working anymore and I tried to update the script to match the new version but I'm stuck.
1) Is there any documentation on the changes made to the code in the latest release?
2) How can I debug the error that's causing the device to show up as unsupported now?
3) If I roll back to the previous release which was partially working and fix the few bugs I had, will I be able to commit those changes to the previous release? will it eventually show up in new releases?

This is what I had partially working:

```
const {} = require('zigbee-herdsman-converters/lib/modernExtend');
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;
const lumi = require('zigbee-herdsman-converters/lib/lumi');
const {
  lumiAction, lumiOperationMode, lumiPowerOnBehavior, lumiZigbeeOTA,
  lumiSwitchType, lumiAirQuality, lumiVoc, lumiDisplayUnit, lumiLight,
  lumiOutageCountRestoreBindReporting, lumiElectricityMeter, lumiPower,
  lumiOverloadProtection, lumiLedIndicator, lumiButtonLock,
} = lumi.modernExtend;



const preventReset = async (type, data, device) => {
  if (
      // options.allow_reset ||
      type !== 'message' ||
      data.type !== 'attributeReport' ||
      data.cluster !== 'genBasic' ||
      !data.data[0xfff0] ||
      // eg: [0xaa, 0x10, 0x05, 0x41, 0x87, 0x01, 0x01, 0x10, 0x00]
      !data.data[0xFFF0].slice(0, 5).equals(Buffer.from([0xaa, 0x10, 0x05, 0x41, 0x87]))
  ) {
      return;
  }
  const payload = {[0xfff0]: {
      value: [0xaa, 0x10, 0x05, 0x41, 0x47, 0x01, 0x01, 0x10, 0x01],
      type: 0x41,
  }};
  await device.getEndpoint(1).write('genBasic', payload, {manufacturerCode});
};


const definition =
    {
      zigbeeModel: ['lumi.switch.rkna01'],
      model: 'ZNXNKG01LM',
      vendor: 'Aqara',
      description: 'Aqara knob H1 (with Neutral)',
      fromZigbee: [fz.lumi_on_off_action, fz.lumi_multistate_action, /*xiaomi.fromZigbee.xiaomi_basic,*/
      lumi.fromZigbee.lumi_basic, lumi.fromZigbee.lumi_specific, fz.lumi_knob_rotation, /*h1_pro_actions,*/ fz.lumi_power],
      toZigbee: [tz.on_off, tz.lumi_switch_operation_mode_opple, tz.aqara_opple_operation_mode],
      meta: {multiEndpoint: true},
      endpoint: (device) => {
          return {'left': 1, 'center': 2, 'right': 3};
      },

      exposes: [
      e.switch().withEndpoint('left'), e.switch().withEndpoint('center'), e.switch().withEndpoint('right'),
      e.device_temperature(),
      e.action(['single', 'double', 'triple', 'hold', 'release', 'start_rotating', 'rotation', 'stop_rotating']),
      e.power_outage_count(), e.energy(), e.voltage(), e.power(),
      exposes.numeric('action_rotation_angle', ea.STATE).withUnit('*').withDescription('Rotation angle'),
      exposes.numeric('action_rotation_angle_speed', ea.STATE).withUnit('*').withDescription('Rotation angle speed'),
      exposes.numeric('action_rotation_percent', ea.STATE).withUnit('%').withDescription('Rotation percent'),
      exposes.numeric('action_rotation_percent_speed', ea.STATE).withUnit('%').withDescription('Rotation percent speed'),
      exposes.numeric('action_rotation_time', ea.STATE).withUnit('ms').withDescription('Rotation time'),
      ],
      onEvent: preventReset,
      configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint1 = device.getEndpoint(1);
        await endpoint1.write('aqaraOpple', {'mode': 1}, {manufacturerCode: 0x115f, disableResponse: true});
      },
};

module.exports = definition;
```



and this is my sad attempt to update it on the latest Z2M release


```
const {} = require('zigbee-herdsman-converters/lib/modernExtend');

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const constants = require('zigbee-herdsman-converters/lib/constants');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
//const tuya = require('zigbee-herdsman-converters/lib/tuya');
//const {} = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;
const lumi = require('zigbee-herdsman-converters/lib/lumi');
const {
  lumiAction, lumiOperationMode, lumiPowerOnBehavior, lumiZigbeeOTA,
  lumiSwitchType, lumiAirQuality, lumiVoc, lumiDisplayUnit, lumiLight,
  lumiOutageCountRestoreBindReporting, lumiElectricityMeter, lumiPower,
  lumiOverloadProtection, lumiLedIndicator, lumiButtonLock,lumiMotorSpeed,
  lumiOnOff, lumiLedDisabledNight, lumiFlipIndicatorLight, lumiPreventReset,
  lumiClickMode, lumiSlider, lumiSetEventMode, lumiSwitchMode,
} = lumi.modernExtend;
const {manufacturerCode} = lumi;


const definition =
    {
      zigbeeModel: ['lumi.switch.rkna01'],
      model: 'ZNXNKG01LM',
      vendor: 'Aqara',
      description: 'Aqara knob H1 (with Neutral)',
      fromZigbee: [fz.on_off, lumi.fromZigbee.lumi_action, lumi.fromZigbee.lumi_action_multistate, /*xiaomi.fromZigbee.xiaomi_basic,*/
      lumi.fromZigbee.lumi_basic, lumi.fromZigbee.lumi_specific, lumi.fromZigbee.lumi_knob_rotation, /*h1_pro_actions,*/ fz.lumi_power],
      toZigbee: [tz.on_off, lumi.toZigbee.lumi_switch_operation_mode_opple, lumi.toZigbee.lumi_switch_power_outage_memory, lumi.toZigbee.lumi_switch_mode_switc],

      meta: {multiEndpoint: true},
      endpoint: (device) => {
          return {'left': 1, 'center': 2, 'right': 3};
      },

      exposes: [
      e.switch().withEndpoint('left'), e.switch().withEndpoint('center'), e.switch().withEndpoint('right'),
      e.device_temperature(),
      e.action(['single', 'double', 'hold', 'release', 'start_rotating', 'rotation', 'stop_rotating']),
      e.power_outage_count(), e.energy(), e.voltage(), e.power(),
      e.enum('operation_mode', ea.ALL, ['event', 'command']).withDescription('Button mode'),
      e.enum('action_rotation_button_state', ea.STATE, ['released', 'pressed']).withDescription('Button state during rotation'),
      e.numeric('action_rotation_angle', ea.STATE).withUnit('*').withDescription('Rotation angle'),
      e.numeric('action_rotation_angle_speed', ea.STATE).withUnit('*').withDescription('Rotation angle speed'),
      e.numeric('action_rotation_percent', ea.STATE).withUnit('%').withDescription('Rotation percent'),
      e.numeric('action_rotation_percent_speed', ea.STATE).withUnit('%').withDescription('Rotation percent speed'),
      e.numeric('action_rotation_time', ea.STATE).withUnit('ms').withDescription('Rotation time'),
      ],
      extend: [lumiZigbeeOTA(), lumiPreventReset()],
};

module.exports = definition;
```

**Reply by Koenkk** [reply]
What error shows up in the z2m log when starting z2m? It will log an error that it cannot load the external converter.

**Reply by mageali** [reply]
The error I get is:

**_Zigbee2MQTT:error 2024-03-15 00:02:20: Failed to load external converter file 'devices/ZNXNKG01LM.js' (lumiPreventReset is not a function)
Zigbee2MQTT:error 2024-03-15 00:02:20: Probably there is a syntax error in the file or the external converter is not compatible with the current Zigbee2MQTT version
Zigbee2MQTT:error 2024-03-15 00:02:20: Note that external converters are not meant for long term usage, it's meant for local testing after which a pull request should be created to add out-of-the-box support for the device_**

I see that "lumiPreventReset " is in lumi.ts and added that file in the lib folder.
Am I missing something?

Sorry, if the question is very basic, I haven't written a line of code for almost 15years

**Reply by Koenkk** [reply]
Can you try with the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) ?

**Reply by mageali** [reply]
I'll take a full backup (using Z2M addon with HAOS) and give it a try.
Sorry first time I learn about Dev Branch...
I have 4 instances of Z2M running on a single HAOS (350+ devices, so I split them), when I install "Edge" I assume I need to stop the corresponding Z2M (the one that is using the same coordinator), correct? anything else I should look out for?

**Reply by Koenkk** [reply]
> I assume I need to stop the corresponding Z2M (the one that is using the same coordinator), correct?

Yes, that's it

**Reply by mageali** [reply]
@Koenkk Thanks for your help... I got it working on the latest release of Z2M, and I tried to do a pull request, but I have some errors (indentation and a couple of lines are too long). I've been staring at the screen trying to figure out how to edit the file again to fix those errors, but I'm lost.

Where should I click to edit the "pull request errors"

**Reply by mageali** [reply]
So I believe it should be fine now. I get the "All checks have passed" message.
Please let me know if this is fine now or anything else needs to be done.

**Reply by nutbox76** [reply]
Hi @mageali ,
Thank you for working on this and all the others that have helped, i bought 2 of these switches in hope they would just work but how wrong i was ,anyway i have it discovered in Z2M edge and online and can control the 3 relays through HA but when the knob on the switch is pressed it activates all 3, the power sensors work but i cant get the rotation to do anything and is null in Z2M.
Is there anything else i need to add to get this working because it was the main reason for buying the switches .
All help gratefully appreciated thank you .

**Comment by nutbox76** [comment]
Can someone tell whether this switch can  dim dum lights or would I also need a dimming relay and program the rotation to use that dimmer relay?
Thank you

**Reply by mageali** [reply]
@nutbox76 
No, it cannot. It is not a dimmer (TRIAC).
It's a physical switch + a knob that happens to be wired.
You would either need a smart bulb or a dimming module (non-smart bulbs).
In either those case, you would need to write the automation to convert the knob rotation to a dimming value for the dimmer/bulb

**Reply by nutbox76** [reply]
I thought as much looks like ive took on more that I can chew lol, Thank you for the info

**Comment by mageali** [comment]
@nutbox76 

> Hi @mageali , Thank you for working on this and all the others that have helped, i bought 2 of these switches in hope they would just work but how wrong i was ,anyway i have it discovered in Z2M edge and online and can control the 3 relays through HA but when the knob on the switch is pressed it activates all 3, 

Yes, as far as I understand that's how Aqara intended it to work. when you press the knob all 3 relays are triggered, but you can control each relay separately from Z2M. I haven't found a way to run it decoupled, and I'm not sure if it's supported by the device. If you have an Aqara hub you can check if there is anyway to run it decoupled from the relay. If there is a way with the hub, then the code could probably be modified to support it also.

> the power sensors work but i cant get the rotation to do anything and is null in Z2M. Is there anything else i need to add to get this working because it was the main reason for buying the switches . All help gratefully appreciated thank you .

In this convertor, you will be able to see the rotation value (action_rotation_angle). of course as soon as you stop rotating it will reset to null, if you specify a state trigger on (action_rotation_angle) you should be able to "catch" the event and the value

**Reply by nutbox76** [reply]
Thank you for your help,unfortunately I don't have a aqara hub but might get one down the road ,I shall have to dig deeper in the code side of things and see what I can learn when I get the time.
